### PR TITLE
Upgrade dependency-check-maven plugin

### PR DESCRIPTION
**Pull request by xiaodong-izettle**
Our current `owasp` build is just passing right now, this should fix the issue. 

But we need to be cautious, as this would break quite some stuff.

No need to hurry, this could wait for your back from vacation.

Ping @nzroller.

### CODE REVIEW

**Comment by adamvoncorswant** [comment]
@xiaodong-izettle : Any reason to not merge this? I suppose that all clients need to bump their dependency version to use this anyway, right?

**Comment by xiaodong-izettle** [comment]
Sure, we could merge this. Yes, that's true. It's just `owasp` build would fail after bumping up, for those repos included this.

**Comment by xiaodong-izettle** [comment]
So feel free to merge. It's just whether we want this to be included with your `checkstyle` stuff or not, @adamvoncorswant . 😄

**Comment by adamvoncorswant** [comment]
added a spell fix as well.

@ngong : care to give this a try, now that you know the drill? :dancer:

File: basepom/pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <buildNumber>${project.version}</buildNumber>
         <checkstyle.version>7.1.2</checkstyle.version>
-        <dependency-check-maven.version>1.4.5</dependency-check-maven.version>
+        <dependency-check-maven.version>2.0.1</dependency-check-maven.version>
```
Comment (adamvoncorswant) [inline basepom/pom.xml]: @xiaodong-izettle : Seems version 2 needs another annotation for suppressions: https://github.com/jeremylong/DependencyCheck#maven

(not that I managed to get https://github.com/iZettle/product-library/pull/342 to build with `Pdep-vuln-check` anyway though, so I'm not sure what to do)

Comment (xiaodong-izettle) [inline basepom/pom.xml]: Yes, probably the old style still works, just deprecated.

**Pull request merged**